timesketch/frontend-ng/src/components/Studio/SigmaRuleModification.vue

```diff
@@ -0,0 +1,257 @@
+    <div class="alertbox" v-if="!isParsingSuccesful">
+      <v-alert dense type="error" outlined>
+        {{ status_text }}
+      </v-alert>
+    </div>
+
+    <v-textarea
+      label="Edit Sigma rule"
+      outlined
+      :color="statusColors()"
+      autocomplete="email"
+      rows="20"
+      v-model="ruleYamlTextArea"
+      background-color="statusColors()"
+      @input="parseSigma(ruleYamlTextArea)"
+      class="editSigmaRule"
+    >
+    </v-textarea>
+
+    <div>
+      <v-btn :disabled="!isParsingSuccesful" @click="addOrUpdateRule(ruleYamlTextArea)" small depressed color="primary">
+        {{ isNewRule ? 'Create Rule' : 'Update Rule' }}
+      </v-btn>
+      <div style="width: 10px; display: inline-block"></div>
+      <v-btn color="primary" text @click="$router.back()"> Cancel </v-btn>
+      <div style="width: 10px; display: inline-block"></div>
+      <v-btn @click="deleteRule(rule_uuid)" small text color="primary" :disabled="isNewRule"
+        ><v-icon>mdi-delete</v-icon></v-btn
+      >
+    </div>
+
+    <div class="alertbox" v-if="isParsingSuccesful">
+      <h3>Preview opensearch query:</h3>
+      <pre>{{ editingRule.search_query }}</pre>
+      <!-- TODO: This should become a global class how to nicely highlight a query -->
+    </div>
+  </v-container>
+</template>
+  
+<script>
+import ApiClient from '../../utils/RestApiClient'
+import { SigmaTemplates, defaultSigmaPlaceholder } from '@/utils/SigmaRuleTemplates'
+import EventBus from '../../main'
+import _ from 'lodash'
+
+export default {
+  props: ['rule_uuid', 'sigmaRule'],
+  data() {
+    return {
+      editingRule: { ruleYaml: defaultSigmaPlaceholder }, // empty state
+      status_chip_text: 'OK',
+      status_text: '',
+      ruleYamlTextArea: {},
+      SigmaTemplates: SigmaTemplates,
+      search: '',
+      isNewRule: false,
+      isUpdatingRule: false,
+      isParsingSuccesful: false,
+    }
+  },
+  watch: {
+    rule_uuid: function (newVal, oldVal) {
+      if (newVal === 'new') {
+        this.editingRule = {
+          title: 'New Sigma Rule',
+        }
+        this.isNewRule = true
+        this.isUpdatingRule = false
+        this.status_chip_text = 'Ok'
+        this.isParsingSuccesful = true
+        this.ruleYamlTextArea = defaultSigmaPlaceholder
+        this.editingRule.rule_yaml = defaultSigmaPlaceholder
+        this.parseSigma(this.editingRule.rule_yaml)
+      } else {
+        this.getRuleByUUID(newVal)
+      }
+    },
+  },
+  loaded() {
+    this.parseSigma(this.editingRule.rule_yaml)
+  },
+  mounted() {
+    if (this.rule_uuid === 'new') {
+      this.editingRule = {
+        title: 'New Sigma Rule',
+      }
+      this.isNewRule = true
+      this.isUpdatingRule = false
+      this.status_chip_text = 'Ok'
+      this.isParsingSuccesful = true
+      this.ruleYamlTextArea = defaultSigmaPlaceholder
+      this.editingRule.rule_yaml = defaultSigmaPlaceholder
+      this.parseSigma(this.editingRule.rule_yaml)
```

Comment:
Please wrap this into a `resetComponent` function, it's the same code as in L106-116

Reply:
done